File: fournisseur-frontend/src/app/services/fournisseur.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Fournisseur } from '../models/fournisseur.model';

@Injectable({
  providedIn: 'root'
})
export class FournisseurService {
  private apiUrl = 'http://localhost:8082/fournisseurs';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Fournisseur[]> {
    return this.http.get<Fournisseur[]>(this.apiUrl);
  }

  get(id: number): Observable<Fournisseur> {
    return this.http.get<Fournisseur>(`${this.apiUrl}/${id}`);
  }

  create(fournisseur: Fournisseur): Observable<Fournisseur> {
    return this.http.post<Fournisseur>(this.apiUrl, fournisseur);
  }

  update(id: number, fournisseur: Fournisseur): Observable<Fournisseur> {
    return this.http.put<Fournisseur>(`${this.apiUrl}/${id}`, fournisseur);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: src/main/java/tn/esprit/fournisseurservice/service/FournisseurService.java
package tn.esprit.fournisseurservice.service;

import tn.esprit.fournisseurservice.entity.Fournisseur;

import java.util.List;

public interface FournisseurService {
    Fournisseur create(Fournisseur fournisseur);
    Fournisseur update(Long id, Fournisseur fournisseur);
    void delete(Long id);
    Fournisseur get(Long id);
    List<Fournisseur> getAll();
}